### DISCUSSION TITLE
[ FF7] Constant flickering and texture warping after 1.8.0

### DISCUSSION THREAD

**Discussion by mechasdasd** [julianxhokaxhiu/FFNx#161] [Q&A] [answered]
(Tested from New Game screen to first savepoint.)

Description:
There is a constant flickering or intermittent black screen. it happens every few seconds in the New Game screen, a few times during the initial cutscene, whenever a character performs an action (like throwing the guards), or just every few steps whenever you walk.
Entering a battle makes all of the textures warp and bug out to the point where it's almost unplayable.

Steps to reproduce:
Start the game

What I have tried:
-1.8.0, 1.8.1 (stable), 1.8.1 (canary)
-1998, 2003 Steam, 7th Heaven updated (all clean installs, all with their respective FFNx versions)
-Following the How to Install guide from here
-Installing to different directories, erasing leftover registry entries, restarting, etc.
-Changing resolutions, enabling vsync, etc. changing settings doesn't do anything (nor does leaving them as is from the zip)
-Modded install with texture mods (also makes no difference whether they're enabled or not)
-Custom and Standard driver settings

What works (sort of):
-Any version below 1.8.0 works fine
-Switching the renderer to DX9 (no screen transitions, some visual effects glitched or missing)
-Switching the renderer to Vulkan (characters render on top of each other, e.g barret over cloud, etc)

Hardware:
Intel I3-3240, AMD R7 200 Series (driver version 20.12.1)

Might take some time but I can upload a video if necessary

**Comment by eve-atum** [comment]
General notes that may or may not help:
- DX9 is known to be buggy in the upstream libraries.  Nothing we can do about that.
- Make sure your AMD drivers are fully updated.  They're all over the place in terms of bugs, so keeping on top of this can help a lot more places than just FF7
- The R7 GPU is six year old tech.  It may say Vulkan on the box, but it is likely to be iffy for actual usage.  Stick to either DX11 (preferred) or OpenGL (usually safe fall-back) on that chip.

**Comment by mechasdasd** [comment]
Yes, I keep my drivers updated as much as I can, as for DX9 and Vulkan, they're the only renderers where I don't get the bug. OpenGL doesn't seem to work in AMD (at least it doesn't on my specific GPU), DX11 has the flickering issue that made me open this thread, and DX12 won't render at all.
If possible I would like to get the DX11 renderer working properly, but I'm not entirely sure the problem is on my end, since DX11 worked fine in previous versions of FFNx.

**Comment by eve-atum** [comment]
I would suggest hopping on a discord (qhimm https://discord.gg/fyE2Vk2 or tsunamods https://discord.gg/xJMxvEk) so we can see your ffnx.toml and ffnx.log files... there may be something instructive in there which isn't easy to copy-paste into a git issue.

**Comment by julianxhokaxhiu** [comment]
Unfortunately there's not much I can do as it's a known AMD driver bug. Please update the to the latest available and if not working, feel free to file a bug to AMD to fix that. See you on Discord.

**Comment by Malmern** [comment]
I currently have the same issue,
it flickers on DX11 and not on Vulkan,
however if I use Vulkan the game crashes at random points.

No other rendering modes are working.

The flicker on DX11 makes the game unplayable. 

I've got an RX580

I'm attaching my current mod list,
maybe one of them is causing the issue?

![Mod list 7th heaven](https://user-images.githubusercontent.com/98621501/151628946-be8ceae0-e6ce-45d5-934e-c13a264351a4.png)

**Reply by julianxhokaxhiu** [reply]
Use DirectX 12, it is a known AMD issue, can't do much about it.

Also see https://github.com/julianxhokaxhiu/FFNx/wiki/FAQ#i-have-an-amd-gpu-and-the-game-has-weird-visual-artifacts-while-playing

**Comment by julianxhokaxhiu** [comment, marked as answer]
For anyone crossing this topic, please consider looking at this link and following steps described: https://github.com/julianxhokaxhiu/FFNx/wiki/FAQ#i-have-an-amd-gpu-and-the-game-has-weird-visual-artifacts-while-playing
